### PR TITLE
Add Configurable Valid Duration Seconds to Rosetta Transaction

### LINKED ISSUE #3856: Rosetta make transaction valid duration configurable with 180s default
### Problem

Mirrornode Rosetta construction API delegate to the go SDK to set the transaction valid duration to 120s by default. For better flexibility, the construction API should support customizing valid duration up to 180s as allowed by the main network and default to 180s. 

### Solution

Support customizing transaction valid duration with default 180s

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:
Add configurable value for Transaction Valid Duration Seconds.
Set the default for Valid Duration Seconds to 180s.

**Related issue(s)**:

Fixes #3856

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rosetta/app/services/construction_service.go [hunk 1]
```diff
@@ -361,21 +371,30 @@ func (c *constructionAPIService) getRandomNodeAccountId() hedera.AccountID {
...
+	return metadataValue, nil
+}
+
+func isValidDuration(validDuration int64) bool {
+	// A value of 0 indicates validDuration is unset
+	if (validDuration >= 0) && (validDuration <= 180) {
+		return true
+	} else {
+		return false
+	}
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/app/services/construction_service.go:397]: nit, can simplify:
```suggestion
	// A value of 0 indicates validDuration is unset
	return validDuration >= 0 && validDuration <= 180
```

File: hedera-mirror-rosetta/app/services/construction_service.go [hunk 2]
```diff
@@ -552,6 +571,22 @@ func transactionSetTransactionId(payer hedera.AccountID, validStartNanos int64)
 	}
 }
 
+func transactionSetValidDuration(validDurationSeconds int64) updater {
+	return func(transaction interfaces.Transaction) *rTypes.Error {
+		if validDurationSeconds == 0 {
+			// Default to 180 seconds
+			validDurationSeconds = 180
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/app/services/construction_service.go:578]: make 180 a constant

File: hedera-mirror-rosetta/app/services/construction_service_test.go [hunk 3]
```diff
@@ -770,6 +807,26 @@ func TestConstructionPayloadsInvalidRequest(t *testing.T) {
...
+			customize: payloadsRequestMetadata(map[string]interface{}{metadataKeyValidDurationSeconds: "-1"}),
+		},
+		{
+			name:      "InvalidValidDurationType",
+			customize: payloadsRequestMetadata(map[string]interface{}{metadataKeyValidDurationSeconds: 120}),
+		},
+		{
+			name:      "InvalidValidDurationType",
+			customize: payloadsRequestMetadata(map[string]interface{}{metadataKeyValidDurationSeconds: 180}),
+		},
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/app/services/construction_service_test.go:825]: duplicate of line 819-821

File: hedera-mirror-rosetta/app/services/construction_service.go [hunk 4]
```diff
@@ -361,21 +371,30 @@ func (c *constructionAPIService) getRandomNodeAccountId() hedera.AccountID {
...
+		if metadataValue, err = tools.ToInt64(value); err != nil || metadataValue < 0 {
+			return metadataValue, errors.ErrInvalidArgument
 		}
 	}
 
-	return validStartNanos, nil
+	return metadataValue, nil
+}
+
+func isValidDuration(validDuration int64) bool {
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/app/services/construction_service.go:391]: perhaps `isValidTransactionValidDuration` is a better name.

File: hedera-mirror-rosetta/app/persistence/testmain_test.go [hunk 5]
```diff
@@ -82,7 +82,7 @@ func addTransaction(
 		Result:               result,
 		TransactionHash:      transactionHash,
 		Type:                 transactionType,
-		ValidDurationSeconds: 120,
+		ValidDurationSeconds: 180,
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/app/persistence/testmain_test.go:85]: nit, here and in transaction_builder.go, we don't have to change the value. these are used in integration tests as the default value. The values are what we persist into db.

Comment (xin-hedera) [inline hedera-mirror-rosetta/app/persistence/testmain_test.go:85]: nit, here and in transaction_builder.go, we don't have to change the value. these are used in integration tests as the default value. The values are what we persist into db.

Comment (xin-hedera) [inline hedera-mirror-rosetta/app/persistence/testmain_test.go:85]: nit, here and in transaction_builder.go, we don't have to change the value. these are used in integration tests as the default value. The values are what we persist into db.

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-rosetta/app/services/construction_service.go (hunk 1 above)
Comment (edwin-greene) [reply hedera-mirror-rosetta/app/services/construction_service.go]: Got it changed.

File: hedera-mirror-rosetta/app/services/construction_service.go (hunk 2 above)
Comment (edwin-greene) [reply hedera-mirror-rosetta/app/services/construction_service.go]: I've added the constant.

File: hedera-mirror-rosetta/app/services/construction_service.go (hunk 4 above)
Comment (edwin-greene) [reply hedera-mirror-rosetta/app/services/construction_service.go]: Sounds good, I've renamed it.

File: hedera-mirror-rosetta/app/persistence/testmain_test.go (hunk 5 above)
Comment (edwin-greene) [reply hedera-mirror-rosetta/app/persistence/testmain_test.go]: Ok, I have reverted those changes.

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**